llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -2088,14 +2088,22 @@ class TargetInstrInfo : public MCInstrInfo {
 
   /// Returns a \p outliner::OutlinedFunction struct containing target-specific
   /// information for a set of outlining candidates. Returns std::nullopt if the
-  /// candidates are not suitable for outlining.
+  /// candidates are not suitable for outlining. \p MinRep is the minimum
+  /// number of times the instruction sequence must be repeated.
   virtual std::optional<outliner::OutlinedFunction> getOutliningCandidateInfo(
       const MachineModuleInfo &MMI,
-      std::vector<outliner::Candidate> &RepeatedSequenceLocs) const {
+      std::vector<outliner::Candidate> &RepeatedSequenceLocs,
+      unsigned MipRep) const {
```

Comment:
```suggestion
      unsigned MinRep) const {
```

Since characters are cheap, can we expand this like `MinRepeates`? Or maybe `MinSequenceCount`?

Reply:
Also, can we do `unsigned MinRep=2` to set a default value instead of creating a new function?

Reply:
I just made `MinRepeates` explicit for `getOutliningCandidateInfo`. I don't want it to be a field/property as TargetInstrInfo as this value can be overridden by the client (either for creating candidates for the local outliner vs. for the global outliner).